core/notifications/src/main/java/com/google/samples/apps/nowinandroid/core/notifications/AndroidSystemNotifier.kt

```diff
@@ -16,17 +16,152 @@
 
 package com.google.samples.apps.nowinandroid.core.notifications
 
+import android.Manifest.permission
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build.VERSION
+import android.os.Build.VERSION_CODES
+import androidx.browser.customtabs.CustomTabsIntent
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationCompat.EXTRA_NOTIFICATION_ID
+import androidx.core.app.NotificationCompat.InboxStyle
+import androidx.core.app.NotificationManagerCompat
 import com.google.samples.apps.nowinandroid.core.model.data.NewsResource
+import dagger.hilt.android.qualifiers.ApplicationContext
 import javax.inject.Inject
 import javax.inject.Singleton
 
+private const val NEWS_NOTIFICATION_REQUEST_CODE = 0
+private const val NEWS_NOTIFICATION_SUMMARY_ID = 1
+private const val NEWS_NOTIFICATION_CHANNEL_ID = ""
+private const val NEWS_NOTIFICATION_GROUP = "NEWS_NOTIFICATIONS"
+
 /**
  * Implementation of [Notifier] that displays notifications in the system tray.
  */
 @Singleton
-class AndroidSystemNotifier @Inject constructor() : Notifier {
+class AndroidSystemNotifier @Inject constructor(
+    @ApplicationContext private val context: Context,
+) : Notifier {
+
+    override fun onNewsAdded(
```

Comment:
```suggestion
    // Post a notification for the given list of news resources.
    override fun post(
```

or maybe `postNotification`?

Reply:
Renamed to `postNewsNotifications` as this interface will also be responsible for creating any other kind of notification in the app.